src/reaktoro_pse/reaktoro_block.py

```diff
@@ -1097,18 +1097,42 @@ def initialize(self):
         _log.info(f"---initializing property block {str(self)}----")
         self.rkt_block_builder.initialize(presolve)
 
-    def deactivate(self):
+    def fix_outputs(self):
+        """Fixes all output variables"""
+        self.output_states = {}
+        for key, obj in self.rkt_solver.output_specs.rkt_outputs.items():
+            pyo_var = obj.get_pyomo_var()
+            pyo_var.pprint()
+            self.output_states[key] = pyo_var.fixed
+            pyo_var.fix()
+
+    def unfix_outputs(self):
+        """Unfixes all output variables"""
+        print(self.output_states)
+        for key, obj in self.rkt_solver.output_specs.rkt_outputs.items():
+            if key in self.output_states:
+                if self.output_states[key] == False:
+                    obj.get_pyomo_var().pprint()
+                    obj.get_pyomo_var().unfix()
+            else:
+                obj.get_pyomo_var().unfix()
+
+    def deactivate(self, fix_outputs=True):
         """Deactivates all constraints and grayboxes"""
         super().deactivate()
+        if fix_outputs:
+            self.fix_outputs()
         for v in self.component_data_objects(Constraint):
             v.deactivate()
-        for v in self.component_data_objects(ExternalGreyBoxModel):
+        for v in self.component_objects(ExternalGreyBoxModel, descend_into=True):
             v.deactivate()
 
-    def activate(self):
+    def activate(self, unfix_outputs=True):
         """Activates all constraints and grayboxes"""
         super().activate()
+        if unfix_outputs:
+            self.unfix_outputs()
         for v in self.component_data_objects(Constraint):
             v.activate()
-        for v in self.component_data_objects(ExternalGreyBoxModel):
+        for v in self.component_objects(ExternalGreyBoxModel, descend_into=True):
```

Comment:
Minor issue here: what happens when a user creates a variable and constraint on the reaktoro block and then uses the activate/deactivate options? From the implementation, it will deactivate everything, including any user-defined constraints. 

Is that the expected behaviour?

This could also potentially then cause DOF issues.

Reply:
Yeah, that is expected behavior. if you deactivate the block you should expect all constraints to deactivatre on it.